R/add_dd.R

```diff
@@ -0,0 +1,183 @@
+          keydate2_dateformat_dd >= cij_end_date_upper &
+          !amended_dates ~ "2D",
+
+        # "2DP"	"Starts in CIJ (allowing +-1 day and ends one day after) - (2DP)"
+        keydate1_dateformat_dd >= cij_start_date_lower &
+          keydate1_dateformat_dd <= cij_end_date_upper &
+          keydate2_dateformat_dd >= cij_end_date_upper &
+          !amended_dates ~ "2DP",
+
+        # "2A"	"Starts in CIJ (Accurate Match after correcting assumed end date) - (2A)"
+        keydate1_dateformat_dd >= cij_start_date &
+          keydate1_dateformat_dd <= cij_end_date &
+          keydate2_dateformat_dd >= cij_end_date &
+          amended_dates ~ "2A",
+
+        # "2AP"	"Starts in CIJ (Accurate Match (allowing +-1 day) after correcting assumed end date) - (2AP)"
+        keydate1_dateformat_dd >= cij_start_date_lower &
+          keydate1_dateformat_dd <= cij_end_date_upper &
+          keydate2_dateformat_dd >= cij_end_date_upper &
+          amended_dates ~ "2AP",
+
+        # "3"	"Ends in CIJ - (3)"
+        keydate1_dateformat_dd <= cij_start_date &
+          keydate2_dateformat_dd >= cij_start_date &
+          keydate2_dateformat_dd >= cij_end_date &
+          !amended_dates ~ "3",
+
+        # "3D"	"Ends in CIJ (starts one day before) - (3D)"
+        keydate1_dateformat_dd <= cij_start_date_lower &
+          keydate2_dateformat_dd >= cij_start_date &
+          keydate2_dateformat_dd >= cij_end_date &
+          !amended_dates ~ "3D",
+
+        # "3DP"	"Ends in CIJ (allowing +-1 day and starts one day before) - (3DP)"
+        keydate1_dateformat_dd <= cij_start_date_lower &
+          keydate2_dateformat_dd >= cij_start_date_lower &
+          keydate2_dateformat_dd >= cij_end_date_upper &
+          !amended_dates ~ "3DP",
+
+        # "4"	"Matches unended MH record - (4)"
+        recid == "04B" &
+          keydate1_dateformat_dd >= cij_start_date &
+          amended_dates ~ "4",
+
+        # "4P"	"Matches unended MH record (allowing -1 day) - (4P)"
+        recid == "04B" &
+          keydate1_dateformat_dd >= cij_start_date_lower &
+          amended_dates ~ "4P",
+
+        # "-" "No Match (We don't keep these)"
+        .default = "-"
+      )
+    )) %>%
+    dplyr::filter(dd_type != "-") %>%
+    dplyr::mutate(smrtype_dd = dplyr::case_when(
+      dd_type %in% c(
+        "1",
+        "1P",
+        "1A",
+        "1AP",
+        "2",
+        "2D",
+        "2DP",
+        "2A",
+        "2AP",
+        "3",
+        "3D",
+        "3DP",
+        "4",
+        "4P"
+      ) ~ "DD-CIJ",
+      dd_type %in% c("no-cij") ~ "DD-No CIJ"
+    )) %>%
+    # tidy up and rename columns to match the format of episode files
+    dplyr::select(
+      chi,
+      recid = recid_dd,
+      keydate1_dateformat = keydate1_dateformat_dd,
+      keydate2_dateformat = keydate2_dateformat_dd,
+      smrtype = smrtype_dd,
+      cij_marker,
+      cij_start_date,
+      cij_end_date,
+      postcode = postcode_dd
+    ) %>%
```

Comment:
We may want to select more columns, to match columns and the format of the episode data

Reply:
There are a lot of 'delay specific' variables we need such as 'delay reason' etc.

Reply:
We also want to 'copy' the other `cij_*` variables to the delay episodes